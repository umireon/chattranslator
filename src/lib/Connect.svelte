<script lang="ts">
  import type { AppContext } from '../../common/constants'
  import type { Firestore } from 'firebase/firestore'
  import type { User } from 'firebase/auth'
  import { generateNonce } from '../service/oauth'
  import { setUserData } from '../service/users'

  export let context: AppContext
  export let db: Firestore
  export let user: User

  async function handleTwitchConnect() {
    const nonce = generateNonce()
    await setUserData(db, user, { nonce })
    const query = new URLSearchParams({
      client_id: context.twitchClientId,
      redirect_uri: `${location.origin}${location.pathname}`.replace(
        /app.html$/,
        'twitch.html'
      ),
      response_type: 'token',
      scope: 'chat:read',
      state: nonce,
    })
    location.href = `https://id.twitch.tv/oauth2/authorize?${query}`
  }
</script>

<main>
  <p>
    <button type="button" on:click={handleTwitchConnect}
      >Connect to Twitch</button
    >
  </p>
</main>
