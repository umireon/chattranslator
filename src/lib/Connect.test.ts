import Connect from './Connect.svelte'
import { render } from '@testing-library/svelte'

test('a', () => {
  const component = render(Connect)
  expect(component.container).toMatchSnapshot()
})
