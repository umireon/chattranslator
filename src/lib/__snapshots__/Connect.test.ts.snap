// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a 1`] = `
<body>
  <div>
    <main>
      <p>
        <button
          type="button"
        >
          Connect to Twitch
        </button>
      </p>
    </main>
  </div>
</body>
`;
